src/components/Player/Player.jsx

```diff
@@ -21,63 +26,82 @@ import {
+  content: () => (<div />),
+}, {
+  name: strings.tab_records,
+  content: (playerId) => (<RecordsPage playerId={playerId} />),
+}, {
+  name: strings.tab_counts,
+  content: (playerId) => (<CountsPage playerId={playerId} />),
+}, {
+  name: strings.tab_histograms,
+  content: (playerId, subInfo) => (<HistogramsPage playerId={playerId} histogramName={subInfo} />),
+}, {
+  name: strings.tab_trends,
+  content: (playerId, subInfo) => (<TrendsPage playerId={playerId} trendName={subInfo} />),
+}, {
+  name: strings.tab_wardmap,
+  content: () => (<div />),
+}, {
+  name: strings.tab_wordcloud,
+  content: () => (<div />),
+}, {
+  name: strings.tab_mmr,
+  content: (playerId) => (<MMRPage playerId={playerId} />),
+}, {
+  name: strings.tab_rankings,
+  content: (playerId) => (<RankingsPage playerId={playerId} />),
+}];
 
-const getPlayerSubroute = (info, playerId, subInfo) => {
-  switch (info) {
-    case 'overview':
-      return <OverviewPage playerId={playerId} />;
-    case 'matches':
-      return <MatchesPage playerId={playerId} />;
-    case 'heroes':
-      return <HeroesPage playerId={playerId} />;
-    case 'pros':
-      return <ProsPage playerId={playerId} />;
-    case 'rankings':
-      return <RankingsPage playerId={playerId} />;
-    case 'histograms':
-      return <HistogramsPage playerId={playerId} histogramName={subInfo} />;
-    case 'peers':
-      return <PeersPage playerId={playerId} />;
-    case 'records':
-      return <RecordsPage playerId={playerId} />;
-    case 'counts':
-      return <CountsPage playerId={playerId} />;
-    case 'trends':
-      return <TrendsPage playerId={playerId} trendName={subInfo} />;
-    case 'mmr':
-      return <MMRPage playerId={playerId} />;
-    default:
-      return <OverviewPage playerId={playerId} />;
-  }
-};
+const playerPagesMapped = (playerId) => playerPages.map(page => ({
+  ...page,
+  route: `/players/${playerId}/${page.name.toLowerCase()}`,
+}));
 
-const Player = ({ params: { accountId, info, subInfo } }) => {
+const Player = ({
+  params: {
+    accountId,
+    info,
+    subInfo,
+  },
+}) => {
   if (!accountId) {
     return <Error />;
   }
-// Need to pass in the action into filter form, need to put that filter form into each subroute as well
+  const defInfo = info || 'overview';
+  // Need to pass in the action into filter form, need to put that filter form into each subroute as well
   return (
     <div>
       <div className={styles.header}>
         <PlayerHeader playerId={accountId} />
         <div style={{ marginTop: 25 }}>
           <TabBar
-            playerId={accountId}
-            activeTab={info}
+            info={defInfo}
             subInfo={subInfo}
             tabs={playerPagesMapped(accountId)}
           />
         </div>
       </div>
-      {getPlayerSubroute(info, accountId, subInfo)}
+      {playerPagesMapped(accountId).filter(page => page.name.toLowerCase() === defInfo).map(page => page.content(accountId, subInfo))}
     </div>
```

Comment:
This feels way more confusing to me than the switch case statement. I'd rather keep it the same way unless there is some sort of reason this is better besides personal preference.